javalib/src/main/scala/java/net/URI.scala

```diff
@@ -135,13 +135,15 @@ final class URI(origStr: String) extends Serializable with Comparable[URI] {
     case _ => false
   }
 
-  def getAuthority(): String = _authority.map(decodeComponent).orNull
+  def getAuthority(): String =
+    _authority.filter(_ != "").map(decodeComponent).orNull
```

Comment:
Very good point. A couple of things use `_authority`. Notably:

- comparison
- `parseServerAuthority`
- `relativize`
- `resolve`